### PR TITLE
Progress toward #2389 and #2390.  This puts the right options in the …

### PR DESCRIPTION
**Pull request by picklesrus**
…menu based on whether a sprite or the stage is selected. It does not update the currently selected options menu item when you change the sprite selector menu yet though.

### Resolves

Progress toward #2389 and #2390

### Proposed Changes

Add a dynamic menu for the sensing of block's options menu.

### CODE REVIEW

**Comment by picklesrus** [comment]
I wonder if I can write a test for this here: scratch-gui/test/integration/blocks.test.js?  But maybe I need to wait til the block auto-changes the selected menu item for that to be worthwhile?

File: src/lib/blocks.js [hunk 1]
```diff
@@ -47,6 +47,32 @@ export default function (vm) {
         };
     };
 
+
+    const jsonForSensingMenus = function (blockMsg, name, menuOptionsFn, colors) {
```
[the remark this answers is not in the document]
Comment (kchadha) [reply src/lib/blocks.js]: We could probably just have this be a one argument function e.g.

menuOptionsFn => {<all the JSON here>}

File: src/lib/blocks.js [hunk 2]
```diff
@@ -185,6 +211,61 @@ export default function (vm) {
...
+                    sensingOfBlock = vm.runtime.flyoutBlocks.getBlock(blockId);
+                    // If we still don't have a block, just return an empty list . This happens during
+                    // scratch blocks construction.
+                    if (!sensingOfBlock) {
+                        return [['', '']];
+                    }
+                    // The block was in the flyout so look up future block info there.
+                    lookupBlocks = vm.runtime.flyoutBlocks;
+                }
+                // There's a block dropped on top of the menu. It'd be nice to evaluate it and
```
Comment (kchadha) [inline src/lib/blocks.js]: Nit: let's move this comment inside the if below instead of above it

**Comment by kchadha** [comment]
@paulkaplan, do you have any ideas on how to test this behavior (e.g. seeing that the menu on the sensing of block dynamically updates based on the value of the second menu on the block)?

**Comment by kchadha** [comment]
@picklesrus, as discussed offline, there seems to be an issue with the toolbox block when switching sprites, e.g. the selected value of the left hand (PROPERTY) menu updates before the selected value of the right hand menu updates, so when switching sprites, the left hand menu gets one step out of sync with the right hand menu.

I think this PR is good to go since addressing this change most probably requires changes in the VM instead of in the gui.

**Pull request merged**